web/src/views/LoginPortal/FirstFactor/FirstFactorForm.tsx

```diff
@@ -117,6 +124,11 @@ export default function (props: Props) {
                         type="password"
                         onKeyPress={(ev) => {
                             if (ev.key === 'Enter') {
+                                if (!username.length) {
+                                    usernameRef.current.focus();
+                                } else if (!password.length) {
+                                    passwordRef.current.focus();
```

Comment:
why not set the password field as errored like for username in that case? That gives a hint about what's wrong.

Reply:
It does, notice how `handleSignIn();` is out-dented, this will cause any fields which are blank to error and focus accordingly.

Reply:
handleSignin will just do it. My bad.